### PR TITLE
bump `@metamask/assets-controllers` to ^13.0.0

### PR DESCRIPTION
**Pull request by jiexi**
## Explanation
Needed to help get us closer to phasing out a single globally selected network.

Bumps `@metamask/assets-controllers` to ^13.0.0 and passes networkClientId arg to newly updated `TokenController.addToken()`, `TokenController.addTokens()`, and `TokenController.watchAsset()`. Adopts changes in other asset-related controllers

* Fixes https://github.com/MetaMask/MetaMask-planning/issues/1020
* See https://github.com/MetaMask/core/pull/1676

## Screenshots/Screencaps

None. Should be no changes apparent to user.

### Via Dapp API
* Visit any [coingecko token page](https://www.coingecko.com/en/coins/ethereum-name-service)
  * On a supported built-in network (switch to mainnet)
* Click the metamask icon on the page
  * Tooltip says: "Add to MetaMask"
* Accept
* Verify the token shows up in wallet UI

### Via Wallet UI
* From wallet ui
* Import new token
* Fill in token details
  * Use `0xc18360217d8f7ab5e7c516566761ea12ce7f9d72` contract address for ENS
* Verify the token shows up in wallet UI

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -103,6 +103,7 @@
     "audit": "yarn npm audit --recursive --environment production --severity moderate"
   },
   "resolutions": {
+    "@metamask/assets-controllers@^9.2.0": "npm:@metamask-previews/assets-controllers@12.0.0-preview.56ca173",
```
Comment (jiexi) [inline package.json:106]: TODO: Rename this PR to `bump @metamask/assets-controllers` after this is merged and released

**Comment by jiexi** [comment]
@metamaskbot update-policies

File: ui/components/app/nfts-items/nfts-items.js [hunk 2]
```diff
@@ -192,7 +192,7 @@ export default function NftsItems({
                 ipfsGateway,
               );
               const nftImageAlt = getNftImageAlt(nft);
-              const isImageHosted = image.startsWith('https:');
+              const isImageHosted = image?.startsWith('https:');
```
Comment (jiexi) [inline ui/components/app/nfts-items/nfts-items.js:195]: I'm uncertain why this is now needed

Comment (jiexi) [reply ui/components/app/nfts-items/nfts-items.js:195]: Seems this was just missed as optional chaining on `image` is also used here

https://github.com/MetaMask/metamask-extension/blob/0139b9913b762adf448d81f48d265a8fbc0abfd3/ui/components/app/nft-details/nft-details.js#L91

going to mark this as resolved

Comment (brad-decker) [reply ui/components/app/nfts-items/nfts-items.js:195]: Unresolving just to confirm that the missing optional chaining seems to have introduced a bug in 11.1.0 that this PR now resolves

**Review by brad-decker — APPROVED** [review]

File: app/scripts/metamask-controller.js [hunk 3]
```diff
@@ -724,17 +738,18 @@ export default class MetamaskController extends EventEmitter {
     this.tokenRatesController = new TokenRatesController(
       {
         chainId: this.networkController.state.providerConfig.chainId,
+        ticker: this.networkController.state.providerConfig.ticker,
```
Comment (adonesky1) [inline app/scripts/metamask-controller.js:741]: what's this change about?

Comment (adonesky1) [reply app/scripts/metamask-controller.js:741]: ah nvm I see it.

Comment (adonesky1) [reply app/scripts/metamask-controller.js:741]: Whoa we're bumping 4 major versions here

**Review by adonesky1 — APPROVED** [review]
LGTM

**Pull request merged**